tests/correction/main.c

```diff
@@ -0,0 +1,45 @@
+#include <mpix/op_correction.h>
+#include <mpix/test.h>
+
+#define WIDTH 16
+#define HEIGHT 16
+
+static uint8_t rgb24frame_in[WIDTH * HEIGHT * 3];
+static uint8_t colorcorrectedframe_out[WIDTH * HEIGHT * 3];
+
+/* uses a 3x3 matrix we will ignore offsets and just use coefficients now */
+static uint8_t ccm_matrix[3 * 3]
+
+void test_correction()
+{
+	int index;
+	for(index = 0; index < HEIGHT * WIDTH * 3; index++)
+	{
+		rgb24frame_in[index] = index;
+		colorcorrectedframe_out[index] = 0;
+	}
+
+	int ccm_index;
+
+	for(ccm_index = 0; ccm_index < 8; ccm_index++)
+	{
+		ccm_matrix[ccm_index] = ccm_index;
+	}
+
+	/*We have an initialized in buffer and a defined out buffer and a ccm_matrix*/
+
+	//call correction
+
+	//walk out_buffer pixel by pixel doing asserts
+
+
+
```

Comment:
Being able to verify correct output is a lot easier once being able to play with the data.

One possible next step is plugging this operation above to the CLI tool, like it was done here for instance:

https://github.com/libmpix/libmpix/blob/2f096866c209f1cfe40cd350e7414f66086d2747/cli/main.c#L415-L438

Reply:
oh I made an array and I am pagin it like the source and asserting, I can look at this too though

Reply:
took a stab at this as well.